controller/cache/cluster.go

```diff
@@ -115,6 +116,13 @@ func (c *clusterInfo) getChildren(obj *unstructured.Unstructured) []appv1.Resour
 	return children
 }
 
+func (c *clusterInfo) isNamespaced(gvk schema.GroupVersionKind) bool {
+	if api, ok := c.apis[gvk]; ok && !api.Namespaced {
+		return false
+	}
+	return true
```

Comment:
Is there any problem here that if we don't see the GVK in the cluster, that we unconditionally consider it as a namespaced? e.g. will deploying an app for the first time that contains a cluster-scoped CRD + cluster-scoped CRD object work okay?

Reply:
good point. working improving it in next PR